File: modules/fully-drawn/android/src/main/java/expo/modules/fullydrawn/FullyDrawnModule.kt
package expo.modules.fullydrawn

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class FullyDrawnModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("FullyDrawn")

    Function("report") {
      val activity = appContext.currentActivity ?: return@Function
      activity.runOnUiThread { activity.reportFullyDrawn() }
    }
  }
}

File: modules/fully-drawn/src/FullyDrawnModule.ts
import { requireOptionalNativeModule } from 'expo-modules-core';

import type { FullyDrawnNativeModule } from './FullyDrawn.types';

const unavailableModule: FullyDrawnNativeModule = {
  report() {},
};

export default requireOptionalNativeModule<FullyDrawnNativeModule>(
  'FullyDrawn',
) ?? unavailableModule;
